rct229/rules/section5/section5rule28.py

```diff
@@ -0,0 +1,169 @@
+                )
+
+                return [
+                    UserBaselineProposedVals(None, subsurface_b, subsurface_p)
+                    for subsurface_b, subsurface_p in proposed_baseline_subsurface_pairs
+                ]
+
+            class UnregulatedSubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule28.BuildingRule.UnregulatedSurfaceRule.UnregulatedSubsurfaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        required_fields={
+                            "$": [
+                                "u_factor",
+                                "solar_heat_gain_coefficient",
+                                "glazed_area",
+                                "opaque_area",
+                            ]
+                        },
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    subsurface_p = context.proposed
+
+                    return {
+                        "baseline_subsurface_u_factor": subsurface_b["u_factor"],
+                        "proposed_subsurface_u_factor": subsurface_p["u_factor"],
+                        "baseline_subsurface_shgc": subsurface_b[
+                            "solar_heat_gain_coefficient"
+                        ],
+                        "proposed_subsurface_shgc": subsurface_p[
+                            "solar_heat_gain_coefficient"
+                        ],
+                        "baseline_subsurface_glazed_area": subsurface_b["glazed_area"],
+                        "proposed_subsurface_glazed_area": subsurface_p["glazed_area"],
+                        "baseline_subsurface_opaque_area": subsurface_b["opaque_area"],
+                        "proposed_subsurface_opaque_area": subsurface_p["opaque_area"],
+                    }
+
+                def rule_check(self, context, calc_vals, data=None):
+                    if (
+                        std_equal(
+                            calc_vals["baseline_subsurface_u_factor"],
+                            calc_vals["proposed_subsurface_u_factor"],
+                        )
+                        and std_equal(
+                            calc_vals["solar_heat_gain_coefficient"],
+                            calc_vals["solar_heat_gain_coefficient"],
+                        )
+                        and std_equal(
+                            calc_vals["glazed_area"], calc_vals["glazed_area"]
+                        )
+                        and std_equal(
+                            calc_vals["opaque_area"], calc_vals["opaque_area"]
+                        )
+                    ):
+                        return True
+                    else:
```

Comment:
We'll have to talk about this `else` part.  I don't understand why the RDS has this part at all. It doesn't seem to add anything more that what is in the rule description. This is what it means for the rule to fail.  So in this case, I am pushing back against the need for raising this warning at all.  When we truly do need such a warning message, I think we should handle it differently.